python/beeai_framework/tools/weather/openmeteo.py

```diff
@@ -75,7 +75,9 @@ def _geocode(self, input: OpenMeteoToolInput) -> dict[str, str]:
         )
 
         response.raise_for_status()
-        results = response.json()["results"]
+        results = response.json().get("results", [])
+        if not results:
+            raise ToolError(f"Location '{input.location_name}' was not found.")
```

Comment:
Is this really the only possible error case if results is empty?

Reply:
`keyerror` error is not very descriptive, this prevents that

Reply:
yes

Reply:
If you call the API with the location that does not exist, then you get an object without the `results` attribute. 
I added a check that fixes that.